init_venv.sh

```diff
@@ -69,45 +92,52 @@ else
   echo torchvision==${TORCHVISION_VERSION} >> ${CONSTRAINTS_FILE}
+  pip install torch==${TORCH_VERSION} torchvision==${TORCHVISION_VERSION} -c ${CONSTRAINTS_FILE} || exit 1
 else
-  pip install torch==${TORCH_VERSION}+cu${CUDA_VERSION_CODE} torchvision==${TORCHVISION_VERSION}+cu${CUDA_VERSION_CODE} -f https://download.pytorch.org/whl/torch_stable.html
   echo torch==${TORCH_VERSION}+cu${CUDA_VERSION_CODE} >> ${CONSTRAINTS_FILE}
   echo torchvision==${TORCHVISION_VERSION}+cu${CUDA_VERSION_CODE} >> ${CONSTRAINTS_FILE}
+  pip install torch==${TORCH_VERSION}+cu${CUDA_VERSION_CODE} torchvision==${TORCHVISION_VERSION}+cu${CUDA_VERSION_CODE} -f https://download.pytorch.org/whl/torch_stable.html \
+          -c ${CONSTRAINTS_FILE} || exit 1
 fi
 
 if [[ -z $CUDA_VERSION_CODE ]]; then
-  pip install --no-cache-dir mmcv-full==${MMCV_VERSION} -f https://download.openmmlab.com/mmcv/dist/cpu/torch${TORCH_VERSION}/index.html -c ${CONSTRAINTS_FILE}
+  pip install --no-cache-dir mmcv-full==${MMCV_VERSION} -f https://download.openmmlab.com/mmcv/dist/cpu/torch${TORCH_VERSION}/index.html -c ${CONSTRAINTS_FILE} || exit 1
 else
-  pip install --no-cache-dir mmcv-full==${MMCV_VERSION} -f https://download.openmmlab.com/mmcv/dist/cu${CUDA_VERSION_CODE}/torch${TORCH_VERSION}/index.html -c ${CONSTRAINTS_FILE}
+  pip install --no-cache-dir mmcv-full==${MMCV_VERSION} -f https://download.openmmlab.com/mmcv/dist/cu${CUDA_VERSION_CODE}/torch${TORCH_VERSION}/index.html -c ${CONSTRAINTS_FILE} || exit 1
 fi
 
 # Install other requirements.
 # Install mmpycocotools and Polygon3 from source to make sure it is compatible with installed numpy version.
-pip install --no-cache-dir --no-binary=mmpycocotools mmpycocotools -c ${CONSTRAINTS_FILE}
-pip install --no-cache-dir --no-binary=Polygon3 Polygon3==3.0.8 -c ${CONSTRAINTS_FILE}
-cat requirements.txt | xargs -n 1 -L 1 pip install --no-cache -c ${CONSTRAINTS_FILE}
+pip install --no-cache-dir --no-binary=mmpycocotools mmpycocotools -c ${CONSTRAINTS_FILE} || exit 1
+pip install --no-cache-dir --no-binary=Polygon3 Polygon3==3.0.8 -c ${CONSTRAINTS_FILE} || exit 1
+cat requirements.txt | xargs -n 1 -L 1 pip install --no-cache -c ${CONSTRAINTS_FILE} || exit 1
 
-pip install -e . -c ${CONSTRAINTS_FILE}
+pip install -e . -c ${CONSTRAINTS_FILE} || exit 1
 MMDETECTION_DIR=`realpath .`
 echo "export MMDETECTION_DIR=${MMDETECTION_DIR}" >> ${venv_dir}/bin/activate
 
-pip install -e $SC_SDK_REPO/src/ote_sdk -c ${CONSTRAINTS_FILE}
-pip install -e $SC_SDK_REPO/src/sc_sdk -c ${CONSTRAINTS_FILE}
-pip install -e $SC_SDK_REPO/src/common/users_handler -c ${CONSTRAINTS_FILE}
-pip install `find $SC_SDK_REPO/.cache -name *.whl`
+
+pip install -e $SC_SDK_REPO/src/ote_sdk -c ${CONSTRAINTS_FILE} || exit 1
+pip install -e $SC_SDK_REPO/src/sc_sdk -c ${CONSTRAINTS_FILE} || exit 1
+pip install -e $SC_SDK_REPO/src/common/users_handler -c ${CONSTRAINTS_FILE} || exit 1
+if [[ -z ${SKIP_SC_SDK_ADDITIONAL_PACKAGES} ]]; then
```

Comment:
where is it used?

Reply:
At the moment SC SDK contains in the subfolder `.cache` the package `e2e-test-framework` with the version `1.172.*`
This version is obsolete -- particularly, it has dependency conflicts with OpenVINO 2021.4 (wrong numpy version, etc)
The conflicts were resolved in the version `1.198.*` of the framework. 

In my own scripts we install the required version of the package `e2e-test-framework` from the inner pip server.
(Please, note that it is the preferred way of installation of this package)
But this requires that the obsolete version of the framework is not installed (otherwise, after this installation numpy package will be reinstalled with a wrong version, etc)
So, we need to have possibility to skip the additional packages' installation.
So, this environment variable is required.

As soon as you and Pavel merge this PR, I will create MR that adds my scripts installing the proper version of `e2e-test-framework` to the infrastructure repository of our CI.

Reply:
I'd suggest making dependency on e2e optional in the first place. Then we'll be able to remove its installation from here at all. @LeonidBeynenson how much effort would it be?